config/sampler_config.go

```diff
@@ -45,12 +47,379 @@ type RulesBasedSamplerCondition struct {
+					return n == conditionValue
+				}
+				return false
+			}
+			return nil
+		case ">":
+			r.Matches = func(spanValue any, exists bool) bool {
+				if n, ok := tryConvertToInt64(spanValue); exists && ok {
+					return n > conditionValue
+				}
+				return false
+			}
+			return nil
+		case ">=":
+			r.Matches = func(spanValue any, exists bool) bool {
+				if n, ok := tryConvertToInt64(spanValue); exists && ok {
+					return n >= conditionValue
+				}
+				return false
+			}
+			return nil
+		case "<":
+			r.Matches = func(spanValue any, exists bool) bool {
+				if n, ok := tryConvertToInt64(spanValue); exists && ok {
+					return n < conditionValue
+				}
+				return false
+			}
+			return nil
+		case "<=":
+			r.Matches = func(spanValue any, exists bool) bool {
+				if n, ok := tryConvertToInt64(spanValue); exists && ok {
+					return n <= conditionValue
+				}
+				return false
+			}
+			return nil
+		}
+	case float64:
+		// check if conditionValue and spanValue are not equal
+		switch condition {
+		case "!=":
+			r.Matches = func(spanValue any, exists bool) bool {
+				if n, ok := tryConvertToFloat(spanValue); exists && ok {
+					return n != conditionValue
+				}
+				return false
+			}
+			return nil
+		case "=":
+			r.Matches = func(spanValue any, exists bool) bool {
+				if n, ok := tryConvertToFloat(spanValue); exists && ok {
+					return n == conditionValue
+				}
+				return false
+			}
+			return nil
+		case ">":
+			r.Matches = func(spanValue any, exists bool) bool {
+				if n, ok := tryConvertToFloat(spanValue); exists && ok {
+					return n > conditionValue
+				}
+				return false
+			}
+			return nil
+		case ">=":
+			r.Matches = func(spanValue any, exists bool) bool {
+				if n, ok := tryConvertToFloat(spanValue); exists && ok {
+					return n >= conditionValue
+				}
+				return false
+			}
+			return nil
+		case "<":
+			r.Matches = func(spanValue any, exists bool) bool {
+				if n, ok := tryConvertToFloat(spanValue); exists && ok {
+					return n < conditionValue
+				}
+				return false
+			}
+			return nil
+		case "<=":
+			r.Matches = func(spanValue any, exists bool) bool {
+				if n, ok := tryConvertToFloat(spanValue); exists && ok {
+					return n <= conditionValue
+				}
+				return false
+			}
+			return nil
+		}
+	case bool:
+		// check if conditionValue and spanValue are not equal
+		switch condition {
+		case "!=":
+			r.Matches = func(spanValue any, exists bool) bool {
+				if n := tryConvertToBool(spanValue); exists && n {
+					return n != conditionValue
+				}
+				return false
+			}
+			return nil
+		case "=":
+			r.Matches = func(spanValue any, exists bool) bool {
+				if n := tryConvertToBool(spanValue); exists && n {
+					return n == conditionValue
+				}
+				return false
+			}
+			return nil
+		}
+	default:
+		return fmt.Errorf("%s value's type must be either string, int, float or bool, but was '%T'", conditionValue, conditionValue)
+	}
+	return nil // what return was missing?
```

Comment:
I think if we made it here there is a problem with a type's inner switch statement.

Reply:
Right, I think I forgot to remove this comment. Good catch.